### PR TITLE
add new protos for resource / policy evaluation history

### PR DESCRIPTION
**Pull request by mrparkers**
also spent some time moving the resource related protos into their own file.

the `PolicyEvaluation` proto is reusing the existing `EvaluatePolicyResult` proto that we will probably revisit and rename in the future.

### CODE REVIEW

File: proto/v1alpha1/rode_policy.proto [hunk 1]
```diff
@@ -171,3 +171,23 @@ message PolicyEntity {
...
+message PolicyEvaluation {
+  // Id represents the unique identifier (UUID) for this particular policy evaluation.
+  string id = 1;
+
+  // ResourceEvaluationId represents the unique identifier (UUID) of the resource evaluation that triggered this policy evaluation.
+  string resource_evaluation_id = 2;
+
+  // Pass represents the overall status for this policy evaluation. This is determined by looking at the result of each rule
+  // within the policy and performing an AND on each one.
+  bool pass = 3;
```
Comment (alexashley) [inline proto/v1alpha1/rode_policy.proto:186]: In `EvaluatePolicy`, at least for the moment, this is populated by a rule called [`pass` in each policy](https://github.com/rode/demo/blob/dceffb2da5c219dd506b7d8ab46777373fa7cd5f/tf/modules/rode/policies/harbor.rego#L3-L5)

https://github.com/rode/rode/blob/17992895bc00d4249f200705f20240d0db8cb4ef/pkg/policy/manager.go#L629

File: proto/v1alpha1/rode_resource.proto [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  // Source represents the source of the resource evaluation request. This should be set by the enforcer or entity performing
+  // the request.
+  ResourceEvaluationSource source = 3;
+
+  google.protobuf.Timestamp created = 4;
+
+  // ResourceVersion represents the specific resource version that was evaluated in this request.
+  ResourceVersion resource_version = 5;
+
+  // PolicyGroupId represents the ID of the policy group that was evaluated in this request.
```
Comment (alexashley) [inline proto/v1alpha1/rode_resource.proto:82]: Should this `policy_group_name`? IIRC there's no id field on `PolicyGroup`, just name and description.

Or we can go the other way and change name to id

Comment (mrparkers) [reply proto/v1alpha1/rode_resource.proto:82]: what about just `policy_group`?

File: proto/v1alpha1/rode_policy.proto (hunk 1 above)
Comment (mrparkers) [reply proto/v1alpha1/rode_policy.proto:186]: Yeah, so `pass` would be duplicated between this and the `EvaluatePolicyResult`.  I probably only need it in one spot.  I think I'll revisit this and make my own proto for the result here, or move some of the fields within `EvaluatePolicyResult` to the top level of `PolicyEvaluation`

File: proto/v1alpha1/rode_resource.proto (hunk 2 above)
Comment (alexashley) [reply proto/v1alpha1/rode_resource.proto:82]: That sounds good to me

**Review by alexashley — APPROVED** [review]

**Pull request merged**